### PR TITLE
Enable listening for branding override updates

### PR DESCRIPTION
**Pull request by stulees**
## Description
Enable being notified of rise-attribute-data _update_ from any other common-template class or component. 

Also providing a convenient method of being notified specifically for branding override portion of attribute data. 

The Branding class at start will hook into being notified of branding override updates. Follow up PR will address applying updates. 

## Motivation and Context
Development of Override Brand Settings

## How Has This Been Tested?
Only unit tests for now. 

## Release Plan:
- As the Submitter, upon requesting review of this pull request, I confirm that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed. 
- As the Reviewer, upon approving the changes in this PR, I confirm I have reviewed and I agree that the [Release Checklist](https://help.risevision.com/hc/en-us/articles/360031119991) has been completed

#### Release Checklist Items Skipped?
n/a

### CODE REVIEW

**Comment by stulees** [comment]
@olegrise Please review, thanks. FYI upon approval this is only merging to a feature branch, not master.

File: src/rise-attribute-data.js [hunk 1]
```diff
@@ -91,12 +99,33 @@ RisePlayerConfiguration.AttributeData = (() => {
...
+    _data !== null && handler( _data );
+  }
+
+  function onBrandingOverride( handler ) {
+    onAttributeData(( attributeData ) => {
+      const brandingOverride = _hasBrandingOverride( attributeData ) ? attributeData.brandingOverride : null;
+
+      handler( brandingOverride );
+    });
+  }
```
Comment (olegrise) [inline src/rise-attribute-data.js:124]: @stulees what do you think of moving [this block](https://github.com/Rise-Vision/common-template/pull/132/files#diff-fc3b84454a3470418aa2eac8259793c1R118-R124) of code to `rise-branding.js`? My logic is that it should be the responsibilities of the component to interpret the data it receives. I like how [_updateComponentsProperties](https://github.com/Rise-Vision/common-template/blob/6b0b7d8dfd37459f072be74039f6a27b682296c4/src/rise-attribute-data.js#L59) method simply passes data to each component without trying to analyze it. I think we should apply the same pattern to the branding component.
To recap, I think branding component should subscribe directly to the `RisePlayerConfiguration.AttributeData.onAttributeData()`.

File: src/rise-attribute-data.js [hunk 2]
```diff
@@ -91,12 +99,33 @@ RisePlayerConfiguration.AttributeData = (() => {
   }
 
   function update( data ) {
-    _updateComponentsProperties( data );
+    _data = JSON.parse( JSON.stringify( data ));
```
Comment (olegrise) [inline src/rise-attribute-data.js:102]: @stulees curious, is there any specific reason why you decided to clone the data object here or just precaution?

File: src/rise-attribute-data.js (hunk 1 above)
Comment (stulees) [reply src/rise-attribute-data.js:124]: @olegrise I don't disagree, we could do this. To clarify my decision making, I chose to do it in rise-attribute-data for two reasons:

- To stay consistent with how we do this in `rise-display-data` and specifically the address data. See https://github.com/Rise-Vision/common-template/blob/master/src/rise-display-data.js#L29-L33. The rise-data-weather component calls this specifically. 
- Since there needs to be a check on the existence of `brandingOverride`, I liked `rise-attribute-data` having this responsibility as it "owns" the data. 

If you think the above aren't reason enough to keep in attribute data then I am good to move responsibility to `rise-branding`. Thoughts?

File: src/rise-attribute-data.js (hunk 2 above)
Comment (stulees) [reply src/rise-attribute-data.js:102]: @olegrise Just being overly cautious when setting reference to the data object. However, we likely don't need to clone. Thoughts?

Comment (olegrise) [reply src/rise-attribute-data.js:102]: @stulees I'd say if you are not really sure what you need this code for, then get rig of it 😉

File: src/rise-attribute-data.js (hunk 1 above)
Comment (olegrise) [reply src/rise-attribute-data.js:124]: @stulees I see. Thanks for the explanation. I'm very sorry, but I strongly feel that this logic should go to rise-branding.

Comment (stulees) [reply src/rise-attribute-data.js:124]: @olegrise Okay I'll make the change, thanks

**Comment by stulees** [comment]
@olegrise Changes have been made, please review again, thanks!
